File: src/org/develar/mapsforgeTileServer/RenderedTile.java
package org.develar.mapsforgeTileServer;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

public final class RenderedTile implements Serializable {
  public final byte[] data;

  public final long lastModified;
  public final String etag;

  public RenderedTile(@NotNull byte[] data, long lastModified, @NotNull String etag) {
    this.data = data;
    this.lastModified = lastModified;
    this.etag = etag;
  }

  public int computeWeight() {
    return data.length + 8 + etag.length();
  }
}

File: server/src/RenderedTile.kt
package org.develar.mapsforgeTileServer

import java.io.Serializable

public class RenderedTile(public val data: ByteArray, public val lastModified: Long, public val etag: String) : Serializable {
  public fun computeWeight(): Int = data.size() + 4 + etag.length()
}
